### DISCUSSION TITLE
Some question about interface

### DISCUSSION THREAD

**Discussion by Drakmour** [zero01101/openOutpaint#182] [Q&A]
1) How to slide the menu? If I open all tabs in extention menu or expand the prompt input window, it goes beneath the extension window, and I can't find a way to slide down. Like in browser or WORD. I can use CTRL + mouse wheel like on canvas but it makes whole A1111 smaller.
2) Is there a way to drag images that are on a canvas? Like in Photoshop there is a default "drag" or "point" instrument. Cuz all other instruments Dream, Img2Img, Mask, Color, Select and Stamp "do serious stuff" and I always afraid to misclick with those.
3) I don't quite understand how History works. It undo some actions, but doesn't some others. When I was testing "select image" I acidentally removed selected part from the image. It was written in the history. I pushed undo. And it just removed the cropped part like I didn't move cropped part, but left the hole in the image. And no way to restore the moved part.
4) I also don't understand how select image works. I select needed part and can move it or do stuff with it, but when I deselect it, it blends with the everything there is on a canvas. If I deselect it while it is on my image, it "overwrites" it and image that was beneath the cropped part is lost. If I drag it on empty part of canvas, I can select it once more and manipulate, but if I rotated and then deselected, no way to grab this piece any more, cuz select only grabs via grid. I guess it is needed so that selected area if moved turn into an object on separate layer and can be "reselected" at any time. Like in photoshop. Cuz right now it works like in old paint.

I dunno if things that I suggest are possible, not an expert in coding or extension making. But more I use it, more questions I have. I hope my questions will not offend the creator. It's not cuz I am so grumpy, just hope it to git better and better, cuz right now it's the best way to inpaint\outpaint in A1111.

**Comment by zero01101** [comment]
hey, sorry for the delay and all, and no offense taken whatsoever lol, questions lead to bugfixes :) 

1: so not trying to be insulting here, but is your resolution like... anachronistically small?  in webUI (with 6x quicksettings making 2 rows taking up a ton of space) at 1280x1024 everything's usable with the largest panel (SD settings) opened - the menu panels can be moved by click-dragging the title bars of course, or you can ctrl+scroll on the webUI interface (not the openOutpaint iframe) to zoom the whole thing out pretty severely 

![image](https://user-images.githubusercontent.com/1649724/214196485-660896a5-fb34-4c21-86f1-52df768ee25c.png)

2: no pointer/object select tool a la photoshop, you'd indeed make a selection with the marquee select tool

3: so most actions are historical, minus masking and canvas expansion specifically; i haven't run across undo not working as expected, but if you find things that are inconsistent/unexpected and reproducible, please by all means file bug reports on those with steps on how to recreate the issue

4: yeah, as mentioned, not quite photoshop lol, but you'd probably be well served by using layers in general; i don't think i'm following what you're doing in that one though?  could you get like a screen recording? i use https://www.cockos.com/licecap/ on windows myself

**Reply by Drakmour** [reply]
> hey, sorry for the delay and all

It's ok, since it's not about some crucial bugs, just a "quality of life" thing. :-)

> 1: so not trying to be insulting here, but is your resolution like... anachronistically small? in webUI (with 6x quicksettings making 2 rows taking up a ton of space) at 1280x1024 everything's usable with the largest panel (SD settings) opened - the menu panels can be moved by click-dragging the title bars of course, or you can ctrl+scroll on the webUI interface (not the openOutpaint iframe) to zoom the whole thing out pretty severely

I have 2560x1080 and use GoogleChrome. And normal 100% size of A1111 that allows to manipulate all A1111 elements and extensions normally looks like on screen, but menu of openOutpaint goes far below the screen.
https://i.imgur.com/6FTO7Gc.png

Yes I can ctrl+wheel to draw view back like on your screen. But I'm no such a good microscope sniper to see what is written on this tiny menu. :-D
https://i.imgur.com/bOLCFKr.png
And even then when you switch tabs back to txt2img you get this.
https://i.imgur.com/MvvDbtv.png
Not very convenient to go adjust % of view each time.

I'll explain other ones a bit later. I'll also try to record it.

**Comment by zero01101** [comment]
>menu of openOutpaint goes far below the screen.

yeah, so like the current fix for that is "close one of the expandable panels for a bit" lol

i wasn't suggesting you _work_ at an absurd zoom level of course, just that you can do so to find an errant title bar or similar... you should also be able to just click the zoom indicator and click reset in chrome, or just press ctrl+0 to zoom back to 100% 

![image](https://user-images.githubusercontent.com/1649724/214443959-22f85319-97aa-403a-898e-52d06e48c894.png)

basically yeah, openOutpaint currently requires a good amount of vertical resolution if you want to have all the panels open at once, that's why then have collapse buttons ;)

**Reply by Drakmour** [reply]
I understand that maybe it will look "lazy" from my side but it's kinda stressful to collapse and expand tabs or manipulate the zoom every time I need to reach the settings menu, cuz it goes beyond the screen even when 1 tab with Stable diffusion settings is opened. :-D 1 tab and it flies away. I just wondered if it is possible to make an option for menu not to beyond the screen but to become vertical scren sized so that the slider will appear to scroll, like it is in prompt tab.
https://i.imgur.com/taEfNbG.png
